src/util/locking/RedisResourceLocker.ts

```diff
@@ -0,0 +1,113 @@
+import type { RedisClient } from 'redis';
+import { createClient } from 'redis';
+import type { Lock } from 'redlock';
+import Redlock from 'redlock';
+import type { ResourceIdentifier } from '../../ldp/representation/ResourceIdentifier';
+import { getLoggerFor } from '../../logging/LogUtil';
+import { InternalServerError } from '../errors/InternalServerError';
+import type { ResourceLocker } from './ResourceLocker';
+
+/**
+ * A locking system that uses a Redis server or any number of
+ * Redis nodes / clusters
+ */
+export class RedisResourceLocker implements ResourceLocker {
+  protected readonly logger = getLoggerFor(this);
+
+  protected readonly clients: RedisClient[];
+  public readonly redlock: Redlock;
+
+  private readonly lockList: Map<string, Lock>;
+
+  public constructor(redisclients: string[]) {
+    this.lockList = new Map();
+    this.clients = [];
+    if (redisclients) {
+      redisclients.forEach((client: string): void => {
+        // Check if portnumber or ip with portnumber
+        // Defenitely not perfect, but configuring this is only for experienced users
+        if (!/(.+:)?[0-9]{4,5}/u.test(client)) {
+          throw new Error(`Invalid data provided to create a Redis client: ${client}`);
+        }
+        const split = client.split(':');
```

Comment:
Actually, replace the `.test` above my a `match` and you already have the two parts.